DataFormats/Detectors/CPV/include/DataFormatsCPV/RawFormats.h

```diff
@@ -62,29 +65,78 @@ class cpvword
   unsigned char bytes[16] = {0};
 };
 
+class cpvheader
+{
+ public:
+  cpvheader() = default;
+  cpvheader(std::vector<char>::const_iterator b, std::vector<char>::const_iterator e)
+  {                                               //reading header from file
+    for (int i = 0; i < 16 && b != e; i++, b++) { //read up to 16 bytes
+      bytes[i] = *b;
+    }
+  }
+  cpvheader(InteractionRecord orbitBC, bool isNoDataExpected, bool isDataContinued)
+  { //writing header
+    //header is 128-bit word.
+    //|127-120|119-112|111-104|103-96|95-88 |87-80 |79-72|71-64|63-56|55-48|47-40|39-32|31-24|23-16|15-8 |7-0  |
+    // byte15  byte14  byte13  byte12 byte11 byte10 byte9 byte8 byte7 byte6 byte5 byte4 byte3 byte2 byte1 byte0
+    //byte = |76543210|
+    bytes[0] = (0x010 & 0x0ff);                                               //bits 11 - 0 trigger id (0x010 = physics trigger)
+    bytes[1] = ((0x010 & 0xf00) >> 8)                                         //bits 11 - 0 trigger id (0x010 = physics trigger)
+               + 0b00100000 * isNoDataExpected + 0b0100000 * isDataContinued; //bit 13 (no data for this trigger) + bit 14 (payload continues from previous page)
+    bytes[2] = (orbitBC.bc & 0x00ff);                                         //bits 27 - 16 bunch crossing
+    bytes[3] = (orbitBC.bc & 0x0f00) >> 8;                                    //bits 27 - 16 bunch crossing
+    bytes[4] = (orbitBC.orbit & 0x000000ff);                                  //bits 63 - 32 orbit
+    bytes[5] = (orbitBC.orbit & 0x0000ff00) >> 8;                             //bits 63 - 32 orbit
+    bytes[6] = (orbitBC.orbit & 0x00ff0000) >> 16;                            //bits 63 - 32 orbit
+    bytes[7] = (orbitBC.orbit & 0xff000000) >> 24;                            //bits 63 - 32 orbit
+    bytes[8] = 0x00;                                                          //bits 64-71 reserved
+    bytes[9] = 0xe0;                                                          //word ID of cpv header (bits 79 - 72)
+    for (int i = 10; i < 16; i++) {
+      bytes[i] = 0; //bits 127-80 must be zeros
+    }
+  }
+  ~cpvheader() = default;
+  bool isOK() const { return (bytes[9] == 0xe0) && (bytes[10] == 0) && (bytes[11] == 0) && (bytes[12] == 0) && (bytes[13] == 0) && (bytes[14] == 0) && (bytes[15] == 0); }
+  bool isNoDataExpected() const { return bytes[1] & 0b00100000; }
+  bool isDataContinued() const { return bytes[1] & 0b0100000; }
+  uint16_t bc() const { return bytes[2] + ((bytes[3] & 0x0f) << 8); }
+  uint32_t orbit() const { return bytes[4] + (bytes[5] << 8) + (bytes[6] << 16) + (bytes[7] << 24); }
+
+ public:
+  unsigned char bytes[16] = {0}; //0 - 127 bits (16 bytes)
```

Comment:
and data member names start with `mXxxx` (camel form). This does not apply to struct, whose data members start with lower case w/o `m`.

Reply:
And "C++ code file names are derived from the class (or the namespace) names". I think classes cpvword, cpvheader, cpvtrailer should extracted from RawFormats.h to separate .h and .cxx files and names are to be capitalized (CpvWord, CpvHeader, CpvTrailer). This is needed for better readability of the code, especially in many years from now.

Reply:
Hello, @shahor02 !
This RawFormats.h was questioning me as well but it was made in this manner before me. So I just decided to follow 
previous schema. @peressounko, is there any idea behind it or I can without worries modify RawFormats.h according to 
common naming schema?

Reply:
Dear @kharlov, all,  I believe, it is better to keep data format structures (not even classes) like  CpvWord, CpvHeader, CpvTrailer in one place where all formats are described then to produce many tiny files. I suggest to keep as is - it is convenient.

Reply:
@shahor02 modification of the code is made because cpv raw format was changed recently. If I understand correctly, 
cpv firmware is already changed and loaded at P2. Previously CPV was acting as trigger detector but now it produces dma pages for every HBF, like a CRU detector. The structure of dma pages was also changed: RDH now just opens new HBF when CPV header & trailers in payload are used to mark triggered events. So I needed to modify RawDecoder and RawWriter accordingly.

Reply:
@sevdokim Ok, thanks for comment. 

For collecting a few classes in one header, I have no strong opinion: I think in case like here, where they are related and always used together, it is permissible.